Linking error USE_GSL=OFF

Built dev branch on takk (gcc 4.8.5) with following cmake flags

```
BUILD_UTKSCAN=ON
USE_GSL=OFF
```

gives this error:

```
Scanning dependencies of target utkscan
Linking CXX executable utkscan
core/CMakeFiles/CoreObjects.dir/DetectorDriver.cpp.o: In function `DetectorDriver::LoadProcessors(Messenger&)':
DetectorDriver.cpp:(.text+0x655f): undefined reference to `FittingAnalyzer::FittingAnalyzer(std::string const&)'
collect2: error: ld returned 1 exit status
make[2]: *** [Scan/utkscan/source/utkscan] Error 1
make[1]: *** [Scan/utkscan/source/CMakeFiles/utkscan.dir/all] Error 2
make: *** [all] Error 2
```

--- a/Analysis/CMakeLists.txt
+++ b/Analysis/CMakeLists.txt
@@ -1,5 +1,6 @@
 option(USE_HRIBF "Use HRIBF library for scan base." OFF)
-option(USE_GSL "Use GSL for Pulse Fitting" OFF)
+CMAKE_DEPENDENT_OPTION(USE_GSL "Compile with GSL" ON "BUILD_UTKSCAN" OFF)
+mark_as_advanced(USE_GSL)
 
 #Check if GSL is installed
 if(USE_GSL)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -50,6 +50,9 @@ if(CMAKE_BUILD_TYPE MATCHES "Debug")
 endif(CMAKE_BUILD_TYPE MATCHES "Debug")
 
 #------------------------------------------------------------------------------
+#We are going to include this additional module here since it could be useful
+#when setting all of the following options.
+include(CMakeDependentOption)
 
 #Install options
 option(BUILD_ACQ "Build and install Acquisition software" ON)

